edx_username_changer/utils.py

```diff
@@ -0,0 +1,142 @@
+"""
+Utility methods for edx-username-changer plugin
+"""
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+
+from social_django.models import UserSocialAuth
+
+from common.djangoapps.student.models import CourseEnrollment
+from openedx.core.djangoapps.django_comment_common.comment_client.utils import (
+    perform_request as perform_forum_request,
+)
+from openedx.core.djangoapps.django_comment_common.comment_client.user import (
+    User as CommentUser,
+)
+from openedx.core.djangoapps.django_comment_common.comment_client.thread import Thread
+from openedx.core.djangoapps.django_comment_common.comment_client.comment import Comment
+
+from edx_username_changer.exceptions import UpdateFailedException
+
+
+User = get_user_model()
+COMMENT_TYPE = "comment"
+THREAD_TYPE = "thread"
+
+
+class EdxUsernameChanger:
+    """
+    Execeptions for edx-username-changer plugin
+    """
+
+    def __init__(self, old_username, new_username):
+        self.old_username = old_username
+        self.new_username = new_username
+
+    def update_username_in_forum(self):
+        """
+        Changes username in Discussion-Forum service
+        """
+        user = User.objects.get(username=self.new_username)
+        comment_user = CommentUser.from_django_user(user)
+        update_comment_user_username(comment_user, self.new_username)
```

Comment:
Shouldn't we perform all these transactions with atomic?

Reply:
It will not help us with anything, as the changes are done through API calls to discussion-forum service.